File: case_studies/m/active_suspension.m
function xdot = dynamics(t, x, u)
    ms = 300; mu = 60; ks = 16000; kt = 190000; bs = 1000;
    zs = x(1); vs = x(2); zu = x(3); vu = x(4);
    xdot = [vs,
            (-ks*(zs-zu) - bs*(vs-vu) + u) / ms,
            vu,
            (ks*(zs-zu) + bs*(vs-vu) - kt*zu - u) / mu];
end

File: test_dynamics/active_suspension.py
import numpy as np

def dynamics(t, x, u):
    ms, mu, ks, kt, bs = 300, 60, 16000, 190000, 1000
    zs, vs, zu, vu = x[0], x[1], x[2], x[3]
    xdot = [vs,
            (-ks*(zs-zu) - bs*(vs-vu) + u) / ms,
            vu,
            (ks*(zs-zu) + bs*(vs-vu) - kt*zu - u) / mu]
    return np.array(xdot)